pkg/datagatherer/k8s/generic.go

```diff
@@ -110,9 +112,41 @@ func (g *DataGatherer) Fetch() (interface{}, error) {
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	// Redact Secret data
+	err = redactList(list)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
 	return list, nil
 }
 
+func redactList(list *unstructured.UnstructuredList) error {
+	// In principal we could only redact the list if it's kind is SecretList or
+	// a generic mixed List, however the test suite does not set the list kind
+	// and it is safer to always check for Secrets.
```

Comment:
@munnerz Does this make sense? Or should I look into the issues with the test suite further?